zulipterminal/model.py

```diff
@@ -355,6 +355,8 @@ def toggle_message_star_status(self, message: Message) -> None:
         base_request = dict(flag='starred', messages=[message['id']])
         if 'starred' in message['flags']:
             request = dict(base_request, op='remove')
+            if message['id'] in self.index['starred_msg_ids']:
+                self.index['starred_msg_ids'].remove(message['id'])
```

Comment:
This would be better occurring in the event handler, as if you have multiple clients running, you want the update in both clients. For example, if you have the webapp and zulip-terminal both open.